samples/animate-images-with-image-overlay/src/main/java/com/esri/arcgismaps/sample/animateimageswithimageoverlay/components/AnimateImagesWithImageOverlayViewModel.kt

```diff
@@ -0,0 +1,200 @@
+import com.esri.arcgismaps.sample.animateimageswithimageoverlay.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import java.io.File
+import java.util.Arrays
+import java.util.Timer
+import kotlin.concurrent.fixedRateTimer
+
+class AnimateImagesWithImageOverlayViewModel(application: Application) : AndroidViewModel(application) {
+
+    private val provisionPath: String by lazy {
+        application.getExternalFilesDir(null)?.path.toString() + File.separator + application.getString(
+            R.string.animate_images_with_image_overlay_app_name
+        )
+    }
+
+    // Get the folder path containing all the image overlays
+    private val filePath = "$provisionPath/PacificSouthWest"
+
+    var opacity by mutableFloatStateOf(1.0f)
+        private set
+
+    fun updateOpacity(opacityFromSlider: Float) {
+        opacity = opacityFromSlider
+        imageOverlay.opacity = opacity
+    }
+
+    val fpsOptions = listOf(60, 30, 15)
+    private val _fps = MutableStateFlow(fpsOptions[1])
+    val fps = _fps.asStateFlow()
+
+    fun updateFpsOption(fpsIndex: Int) {
+        _fps.value = fpsOptions[fpsIndex]
+    }
+
+    var isStarted by mutableStateOf(true)
+        private set
+
+    fun updateIsStarted(isStartedFromButton: Boolean) {
+        isStarted = isStartedFromButton
+        toggleAnimationTimer()
+    }
+
+
+    // Create an envelope of the pacific southwest sector for displaying the image frame
+    private val pointForImageFrame = Point(
+        x = -120.0724,
+        y = 35.1310,
+        spatialReference = SpatialReference.wgs84()
+    )
+    private val pacificSouthwestEnvelope = Envelope(
+        center = pointForImageFrame,
+        width = 15.0958,
+        height = -14.3770
+    )
+
+    // Create a scene with the dark gray basemap and elevation source
+    val arcGISScene by mutableStateOf(ArcGISScene(BasemapStyle.ArcGISDarkGray).apply {
+        // Create a camera, looking at the pacific southwest sector
+        val observationPoint = Point(-116.621, 24.7773, 856977.0)
+        val camera = Camera(observationPoint, 353.994, 48.5495, 0.0)
+        initialViewpoint = Viewpoint(pacificSouthwestEnvelope, camera)
+    })
+
+    // Create a message dialog view model for handling error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    // Keep track of the list of image frames added in cache
+    private var imageFrames = mutableListOf<ImageFrame>()
+
+    // Keep track of the image frame currently in view
+    private var imageFrameIndex = 0
+
+    // Timer task to customize frame rates
+    private var timer: Timer? = null
+
+    var imageOverlay = ImageOverlay()
+        private set
+
+    init {
+
+        // Get the image files from local storage as an unordered list
+        (File(filePath).listFiles())?.let { imageFiles ->
```

Comment:
you might be able to simplify the logic a bit by calling `sort` and `forEach` here too
```suggestion
        (File(filePath).listFiles()?.sorted()?.forEach { file ->
```